### DISCUSSION TITLE
Binding to focus widget?

### DISCUSSION THREAD

**Discussion by TomJGooding** [Textualize/textual#1721] [Q&A] [answered]
How might you add a binding to focus a particular widget?

I've checked the documentation and see there's`set_focus()` and `focus()` methods available, but I'm struggling trying implement these with a binding in the app.

For example. let's say I wanted to add a vim-like binding to `i` which would focus the form input widget:

```python
from textual.app import App, ComposeResult
from textual.widgets import Footer, Input, Static


class ExampleForm(Static):
    def compose(self) -> ComposeResult:
        yield Input(placeholder="Enter text here...")


class ExampleApp(App):
    BINDINGS = [("q", "quit", "Quit"), ("i", "NotImplemented", "Focus Form")]

    def compose(self) -> ComposeResult:
        yield ExampleForm()
        yield Footer()

    def on_input_submitted(self) -> None:
        self.query_one(Input).value = ""

```

Many thanks in advance for any advice on this!

**Comment by davep** [comment, marked as answer]
You're almost there with the content of your `on_input_submitted`. As you note, widgets have a `focus` method, so if want a binding to focus your `Input` widget you could have something like:

```python
...
    BINDINGS = [ ..., ("i", "focus_input", "Focus Form" ) ]
...
    def action_foocus_input( self ) -> None:
        self.query_one(Input).focus()
```

Of course, if you end up having more than one `Input` on your screen you'll want to narrow it down, perhaps using an `id` for the `Input` you want to focus.

**Reply by TomJGooding** [reply]
Many thanks Dave. I feel a bit silly now, think I must have had blinders on and was trying to only define a function in the binding itself.

**Reply by davep** [reply]
No worries, there's a lot of parts to get to grips with. Glad to have helped. :-)
